src/gtp_path.erl

```diff
@@ -370,7 +371,9 @@ peer_state_change(_, #peer{state = State}, #{reg_key := RegKey}) ->
 %%% Internal functions
 %%%===================================================================
 
-enter_peer_state_action(State, Data) ->
+enter_peer_state_action(State, State, _Data) ->
+    [];
+enter_peer_state_action(_, State, Data) ->
```

Comment:
Looks like this clause will create warning like `this clause X cannot match because a previous clause at line N always matches`.

Reply:
It didn't happen when I or the CI build it and I really can't see how this should happen.

Reply:
It strange, looks like `rebar` somehow ignores such functions, but such a construction should have created a warning.

Reply:
You are right, everything is correct. I just poorly looked at the code for pattern matching.